src/Client.ts

```diff
@@ -208,16 +208,17 @@ export default class Client {
     let recipientKey: PublicKeyBundle | null = null
     const stream = this.apiClient.queryIterator(
       { contentTopics: [buildUserContactTopic(peerAddress)] },
-      { pageSize: 5 }
+      { pageSize: 5, direction: SortDirection.SORT_DIRECTION_DESCENDING }
```

Comment:
I think this is a bug, we want the freshest bundles right?

Reply:
IIRC @jazzz was concerned about people spamming the topic

Reply:
We should be able to mitigate that by filtering out bogus bundles

Reply:
@mkobetic Early on I did not want the freshest bundle.
- Migration to ContactBundlesV1 ran into the issue that older clients could not read the new bundle format until they upgraded.
- Currently the ContactBundle only changes when the bundle format changes. Yet it is published on every client initialization.
- To simplify reading, and negate spam issues it was easiest to read the first bundle only, rather than sorting through pages to find the most recent compatible bundle. 
- This also lead to a consistent experience between different versioned clients.

As the bundles are substantively different in V2, you are correct we'll need to handle this properly via filtering or adding a retry on ContactBundle Decoding. 

I'd suggest now is the time for a model where the contactBundle is only published if needed.

Reply:
+1 on only publishing if needed. I must have hundreds in my topic

Reply:
I agree with fixing the publishing, but I think it could be argued it's out of scope for this PR. Do you see it as a gating factor for this PR?

I can definitely put more effort into making the filtering more robust. Would that be satisfactory?

Reply:
That seems reasonable for this PR. We can handle the publishing frequency when we switch to publish V2 bundles

Reply:
I double checked, we are [filtering out invalid signatures](https://github.com/xmtp/xmtp-js/pull/166/files#diff-b7961e132a9ada7fe5b4cdec69198fc7e4fe506e45b5a7fe29f399a01c45ff53R221), so I think we should be good to go in this regard.

WRT to spamming one's contact topic with junk, republishing valid bundle frequently is probably actually helpful with avoiding having to page through many pages of garbage to get to a valid bundle. When we go fix that we may need to look at checking the contact topic regularly to see what's on top, rather than just at client start time.